common/src/androidTest/kotlin/JSONArrayBugInstrumentedTest.kt

```diff
@@ -0,0 +1,33 @@
+import android.os.Build
+import com.revenuecat.purchases.common.networking.MapConverter
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+
+//@org.junit.runner.RunWith(androidx.test.ext.junit.runners.AndroidJUnit4::class)
+class JSONArrayBugInstrumentedTest {
+    @org.junit.Test
+    fun testReproduceIncorrectJSONArray() {
+        // these tests should only be run on Android 4, since that's the version that has the bug.
+        val osLevel = Build.VERSION.SDK_INT
+        if (!(osLevel >= Build.VERSION_CODES.ICE_CREAM_SANDWICH && osLevel <= Build.VERSION_CODES.KITKAT)) {
+            return
```

Comment:
I added this bit to limit the test to android 4, let me know if there's a better way to handle it

Reply:
Hmm I don't think there is a better way... Possibly don't need to check >= than `ICE_CREAM_SANDWICH` since that's the min version we support.